tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/websocket/RetryClientTest.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http:www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http:www.apache.orglicensesLICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specif ic language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.test.service.websocket;
+
+import org.ballerinalang.test.context.BallerinaTestException;
+import org.ballerinalang.test.util.websocket.client.WebSocketTestClient;
+import org.ballerinalang.test.util.websocket.server.WebSocketRemoteServer;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.net.URISyntaxException;
+import java.nio.ByteBuffer;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This Class tests the retry support of the WebSocket client and server.
+ */
+@Test(groups = {"websocket-test"})
+public class RetryClientTest extends WebSocketTestCommons {
+
+    private WebSocketRemoteServer remoteServer;
+    private String url =  "ws://localhost:21030";
+    private int port = 15300;
+    private String textSent = "hi all";
+
+    @Test(description = "Tests the retry function using the WebSocket client (starting the server " +
+            ", sending, and receiving text frames.  After that, restart that server and do the same.)")
+    public void testRetry() throws URISyntaxException, InterruptedException, BallerinaTestException {
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        remoteServer = new WebSocketRemoteServer(port);
+        remoteServer.run();
+        WebSocketTestClient client = new WebSocketTestClient(url);
+        client.handshake();
+        client.setCountDownLatch(countDownLatch);
```

Comment:
We don't need a countdownLatch for handshake? Can you please go through the code and try to understand the purpose of countDownLatch. You can check other test cases as well

Reply:
I didn't set the countdownLatch for waiting  to finish the handshake. Here, I set the countdownLatch into the webSocketHandler. Other testcases also set this:
https://github.com/ballerina-platform/ballerina-lang/blob/master/tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/websocket/SimpleProxyTestCommons.java#L77